commands/operator-sdk/cmd/generate/internal/genutil.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package genutil
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/operator-framework/operator-sdk/pkg/scaffold"
+)
+
+func BuildCodegenBinaries(genDirs []string, binDir, codegenSrcDir string) error {
+	for _, gd := range genDirs {
+		err := runGoBuildCodegen(binDir, codegenSrcDir, gd)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func runGoBuildCodegen(binDir, repoDir, genDir string) error {
+	binPath := filepath.Join(binDir, filepath.Base(genDir))
+	installCmd := exec.Command("go", "build", "-o", binPath, genDir)
+	installCmd.Dir = repoDir
+	isVerbose := false
+	if gf, ok := os.LookupEnv("GOFLAGS"); ok && len(gf) != 0 {
+		installCmd.Env = append(os.Environ(), "GOFLAGS="+gf)
+		if strings.Contains(gf, "-v") {
+			isVerbose = true
+		}
+	}
+	if isVerbose {
+		installCmd.Stdout = os.Stdout
+		installCmd.Stderr = os.Stderr
+	} else {
+		installCmd.Stdout = ioutil.Discard
+		installCmd.Stderr = ioutil.Discard
+	}
```

Comment:
I'm seeing the same output for both verbose and non-verbose mode:
```console
$ GOFLAGS="-v" operator-sdk generate openapi
INFO[0001] Running OpenAPI code-generation for Custom Resource group versions: [app:[v1alpha1], ]
2019/01/25 15:34:19 Code for OpenAPI definitions generated
INFO[0004] Create deploy/crds/app_v1alpha1_appservice_crd.yaml
INFO[0004] Code-generation complete.

$ operator-sdk generate openapi
INFO[0001] Running OpenAPI code-generation for Custom Resource group versions: [app:[v1alpha1], ]
2019/01/25 15:36:12 Code for OpenAPI definitions generated
INFO[0004] Create deploy/crds/app_v1alpha1_appservice_crd.yaml
INFO[0004] Code-generation complete.
```

Does verbose mode output something extra if the install fails?

Reply:
Are you building the SDK with go 1.11?

Reply:
If the installation or generation isn't working, `GOFLAGS="-v"` will print more information. Otherwise output will be the same.

Reply:
I've got the same output when I pass in an invalid `types.go`  to `generate openapi`:
```Go
type AppServiceSpec struct {
	Size abcd `json:"size"`
}
```
```console
$ GOFLAGS="-v" operator-sdk generate openapi
INFO[0000] Running OpenAPI code-generation for Custom Resource group versions: [app:[v1alpha1], ]
E0127 17:29:23.092281   96805 openapi.go:365] Error when generating: size, Size invalid type
2019/01/27 17:29:23 OpenAPI code generation error: Failed executing generator: some packages had errors:
cannot generate spec for type invalid type
Error: failed to perform code-generation: failed to exec []string{"/Users/haseeb/work/go-space/src/github.com/example-inc/app-operator/build/_output/bin/openapi-gen", "--input-dirs", "github.com/example-inc/app-operator/pkg/apis/app/v1alpha1", "--output-package", "github.com/example-inc/app-operator/pkg/apis/app/v1alpha1", "--output-file-base", "zz_generated.openapi", "--go-header-file", "/Users/haseeb/work/go-space/src/github.com/example-inc/app-operator/vendor/k8s.io/gengo/boilerplate/boilerplate.go.txt"}: exit status 1

$ operator-sdk generate openapi
INFO[0000] Running OpenAPI code-generation for Custom Resource group versions: [app:[v1alpha1], ]
E0127 17:29:29.721581   96815 openapi.go:365] Error when generating: size, Size invalid type
2019/01/27 17:29:29 OpenAPI code generation error: Failed executing generator: some packages had errors:
cannot generate spec for type invalid type
Error: failed to perform code-generation: failed to exec []string{"/Users/haseeb/work/go-space/src/github.com/example-inc/app-operator/build/_output/bin/openapi-gen", "--input-dirs", "github.com/example-inc/app-operator/pkg/apis/app/v1alpha1", "--output-package", "github.com/example-inc/app-operator/pkg/apis/app/v1alpha1", "--output-file-base", "zz_generated.openapi", "--go-header-file", "/Users/haseeb/work/go-space/src/github.com/example-inc/app-operator/vendor/k8s.io/gengo/boilerplate/boilerplate.go.txt"}: exit status 1
```
Do you have an example where it differs for you?

In any case if there's no extra output for verbose mode when the command succeeds then we definitely want it to be verbose for when it fails. So I would recommend we should always keep this verbose and print everything out to the console.